packages/core/src/features/css.js

```diff
@@ -66,9 +70,11 @@ export const createCssFunction = (/** @type {Config} */ config, /** @type {Sheet
 	})
 
 /** Creates a composer from a configuration object. */
-const createComposer = (/** @type {InitComposer} */ { variants: initSingularVariants, compoundVariants: initCompoundVariants, defaultVariants: initDefaultVariants, ...style }, /** @type {Config} */ config) => {
+const createComposer = (/** @type {InitComposer} */ { variants: initSingularVariants, compoundVariants: initCompoundVariants, defaultVariants: initDefaultVariants, componentName, ...style }, /** @type {Config} */ config) => {
+	/** @type {string} */
+	const hash = componentName ? `${componentName}-${toHash(style)}` : toHash(style)
```

Comment:
I believe `componentName` is not a part of the hash.
It probably should be something like
```ts
const className = toTailHash(config.prefix) + ['c', componentName, toHash(style)].filter(Boolean).join('-');
```

Reply:
_If_ I understand you correctly (and I'm not sure I do), I think both pieces of code are doing the same thing; it's just my use of a variable name `hash` that is causing confusion.

My code is equivalent to doing this (note the change from `hash` to `baseClass`):
```JS
let baseClass = toHash(style)
if (componentName) {
  baseClass = `${componentName}-${baseClass}`
}
const className = `${toTailDashed(config.prefix)}c-${baseClass}`
```
So in both your code and mine, the `componentName` is getting conditionally added after the prefix and `c-`, but before the generated hash.

Reply:
But I'll change the name of the variable, to avoid confusion. Let me know if you're okay with that.

Reply:
Yep, they're doing the same thing, my note was about naming.
Thanks!